torch/distributed/_tensor/dispatch.py

```diff
@@ -159,7 +159,7 @@ def _operator_dispatch(
     # tensors before calling the local op
     assert output_sharding.schema_suggestions is not None
     suggested_input_schema = output_sharding.schema_suggestions[0]
-    needs_redistribute = suggested_input_schema is not op_schema
+    needs_redistribute = suggested_input_schema != sharding_propagator.prepare_op_schema(op_call, args, kwargs)
```

Comment:
Somehow the `op_schema` got changed during sharding_prop and it's why `test_index` failed.

Reply:
re-do prepare_schema seems pretty hacky.. possibly because index op adjust the op schema. We should do some serious rewrite later.

Reply:
or maybe is it ok to disable index op test for now?

Reply:
Nah I think we still want index op to work, let's land as is and we can figure out how to remove the redundant prepare_schema later.